docs/developer/extending/apps/key-concepts.mdx

```diff
@@ -145,6 +146,193 @@ FIFO queues (ending in `.fifo`) must be configured to have `ContentBasedDeduplic
+:::
+
+See the example payload sent from Saleor for these payment events: [Payment actions payload](/developer/extending/apps/sample-webhook-payloads.mdx#payment-actions-payload).
+
+For payments that require confirmation (such as 3D Secure) Saleor triggers two events:
+- `PAYMENT_PROCESS` - when the `checkoutComplete` mutation is run for the first time, payment is created and waiting for the user's confirmation.
+- `PAYMENT_CONFIRM` - when the `checkoutComplete` mutation is run for the second time, payment is confirmed.
+
+For payments without a confirmation step, Saleor triggers only the `PAYMENT_PROCESS` event.
+
+##### Expected response
+
+Below is the list of fields that webhook can return in the JSON response for a payment action. All fields are optional:
+
+- `action_required` - (boolean) determines if additional action (confirmation) is required.
+- `action_required_data` - (object) additional data that may be needed to confirm the payment, for example, URL to redirect the user to the confirmation page in the frontend.
+- `amount` - (number) transaction amount; use this field only if the charged amount is different than checkout's total.
+- `customer_id` - (string) gateway's customer ID.
+- `error` - (string) error message; if provided, the transaction will be marked as failed in Saleor and payment will have to be retried. Use this field if you need to return any errors.
+- `payment_method` - (object) information about payment method chosen by the user.
+- `psp_reference` - (string) reference which can be used to find a payment object in Saleor, once the order is created.
+- `transaction_id` - (string) gateway's transaction ID
+- `transaction_already_processed` - (boolean) use this field to mark a transaction as already processed in your gateway
+
+The response fields match with the `GatewayResponse` data class used in Saleor to represent a response from a payment gateway; you can read more about these fields [here](/developer/extending/payment-gateways.mdx#gatewayresponse).
+
+Example response for the `PAYMENT_PROCESS` event when confirmation is required:
+
+```json
+{
+   "action_required": true,
+   "action_required_data":{"confirmation_url": "https://www.example.com/3ds-confirmation/"},
+   "customer_id":"customer-1234",
+   "payment_method": {
+     "brand":"Visa",
+     "exp_month":"01",
+     "exp_year":"2025",
+     "last_4":"4242",
+     "name":"John Doe",
+     "type":"Credit card",
+   },
+   "transaction_id":"transaction-1234"
+}
+```
+
+#### Use payment webhook's payment method in checkout
+
+If payment webhooks are configured properly, you can use their payment methods in checkout. To do this, you need to [list available payment methods for your checkout or shop](/developer/extending/apps/key-concepts.mdx#listing-all-available-payment-gateways). Each payment method from a payment webhook has an ID of this format: `app:<app-database-id>:<id-returned-by-webhook>`
```

Comment:
Enabling webhook is required? AFAIR by default new webhooks created in dashboard are not enabled

Reply:
Yes, webhook needs to be active.